### ISSUE TITLE
`prefer-regexp-test`: should catch `string.search(regex) !== -1`

### ISSUE THREAD

**Issue by dimaMachina** [sindresorhus/eslint-plugin-unicorn#2518]
### Description

I think this rule can catch `String#search` with `!== -1` cases

### Fail

```js
string.search(regex) !== -1
```

### Pass

```js
string.search(regex)
```

**Comment by jacks-sam1010** [comment]
Hello, we tried to solve the issue.

This is what we did:

This change adds support for catching String#search with !== -1 cases in the prefer-regexp-test rule. It includes modifications to the rule implementation, tests, and documentation.

You can review changes in this commit: https://github.com/jacks-sam1010/sindresorhus-eslint-plugin-unicorn-2518/commit/7ed2e7b64a2ea28c8273b830dace7b367646b323.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by sindresorhus** [comment]
> string.search(regex)

Did you mean `regex.test(string)`?

**Comment by dimaMachina** [comment]
nope https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search
